### PR TITLE
bugfix/redis connection timeouts - ALT

### PR DESCRIPTION
**Pull request by aryszka**
branched from https://github.com/zalando/skipper/pull/1592

- [x] upgrade redis client, v7 -> v8
- [x] test dropping the pipe from zremscore+zcard

Signed-off-by: Arpad Ryszka <arpad.ryszka@gmail.com>

### CODE REVIEW

**Comment by szuecs** [comment]
:+1:

**Comment by aryszka** [comment]
:+1:

File: ratelimit/redis.go [hunk 1]
```diff
@@ -243,21 +245,26 @@ func (c *clusterLimitRedis) AllowContext(ctx context.Context, clearText string)
 		return false
 	}
 
-	pipe := c.ring.TxPipeline()
```
Comment (AlexanderYastrebov) [inline ratelimit/redis.go:246]: What is the point of dropping pipeline? Would not it double the number of roundtrips?

Comment (szuecs) [reply ratelimit/redis.go:246]: Great question!
The problem was, that this created socket writes from client side to a socket which was already half closed, so the Linux kernel which hosted redis just sent back RSTs.
So TxPipeline is network unsafe.

Comment (AlexanderYastrebov) [reply ratelimit/redis.go:246]: As a note, the ring implementation [has changed quite a lot](https://github.com/go-redis/redis/compare/v7.0.0-beta.4...v8.3.3#diff-6b4fe479e7ce902cf98e5e4657ad83cba6afca1cd1703d99fd0f446dd53081a0)

**Comment by AlexanderYastrebov** [comment]
@aryszka @szuecs I have noticed significant performance improvement (at least on local setup):
```
$ git checkout v0.11.186 && make && bin/skipper -inline-routes='* -> clusterRatelimit("baseline-rps", 1, "10m") -> status(200) -> <shunt>' -access-log-disabled -enable-ratelimits -enable-swarm -swarm-redis-urls=localhost:6379
$ echo 'GET http://localhost:9090/' | vegeta attack -duration=1m -rate=0 --max-workers=100 | vegeta report
Requests      [total, rate, throughput]         28908, 481.79, 1.66
Duration      [total, attack, wait]             1m0s, 1m0s, 131.785ms
Latencies     [min, mean, 50, 90, 95, 99, max]  124.474ms, 207.847ms, 216.345ms, 219.481ms, 220.441ms, 223.001ms, 288.77ms
Bytes In      [total, mean]                     518544, 17.94
Bytes Out     [total, mean]                     0, 0.00
Success       [ratio]                           0.35%
Status Codes  [code:count]                      200:100  429:28808  
Error Set:
429 Too Many Requests
```
vs 
```
$ git checkout v0.11.188 && make && bin/skipper -inline-routes='* -> clusterRatelimit("baseline-rps", 1, "10m") -> status(200) -> <shunt>' -access-log-disabled -enable-ratelimits -enable-swarm -swarm-redis-urls=localhost:6379
$ echo 'GET http://localhost:9090/' | vegeta attack -duration=1m -rate=0 --max-workers=100 | vegeta report
Requests      [total, rate, throughput]         685599, 11426.67, 0.07
Duration      [total, attack, wait]             1m0s, 1m0s, 15.837ms
Latencies     [min, mean, 50, 90, 95, 99, max]  453.376µs, 8.382ms, 7.871ms, 12.569ms, 14.322ms, 18.501ms, 41.228ms
Bytes In      [total, mean]                     12340710, 18.00
Bytes Out     [total, mean]                     0, 0.00
Success       [ratio]                           0.00%
Status Codes  [code:count]                      200:4  429:685595  
Error Set:
429 Too Many Requests
```

I seems to be caused by v7 to v8 upgrade as numbers in https://github.com/zalando/skipper/pull/1590 were moderate and [improved significantly](https://github.com/zalando/skipper/pull/1590#issuecomment-727092844) as well.

**Pull request merged**